pkg/kdump/kdump-view.jsx

```diff
@@ -20,166 +20,198 @@
-                                   onChange={value => this.props.onChange("sshkey", value)} />
-                    </FormGroup>
+        if (storageLocation === "ssh") {
+            newSettings.targets.ssh.server = server;
+            newSettings.targets.ssh.sshkey = sshkey;
+        }
 
-                    <FormGroup fieldId="kdump-settings-ssh-directory" label={_("Directory")}>
-                        <TextInput id="kdump-settings-ssh-directory" key="directory"
-                                   placeholder="/var/crash" value={directory}
-                                   data-stored={directory}
-                                   onChange={value => this.props.onChange("path", value)} />
-                    </FormGroup>
-                </>
-            );
+        if (storageLocation === "nfs") {
+            newSettings.targets.nfs.server = server;
+            newSettings.targets.nfs.export = exportPath;
         }
 
-        const targetDescription = {
-            local: _("Local filesystem"),
-            nfs: _("Remote over NFS"),
-            ssh: _("Remote over SSH"),
-        };
-        // we don't support all known storage options currently
-        const storageDest = this.state.storeDest;
-        return (
+        handleSave(newSettings)
+                .then(Dialogs.close)
+                .finally(() => setIsSaving(false))
+                .catch(error => {
+                    if (error.details) {
+                        // avoid bad summary like "systemd job RestartUnit ["kdump.service","replace"] failed with result failed"
+                        // if we have a more concrete journal
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error.details))));
+                    } else {
+                        // without a journal, show the error as-is
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error))));
+                    }
+                });
+    };
+
+    return (
+        <Modal position="top" variant="small" id="kdump-settings-dialog" isOpen
+               title={_("Crash dump location")}
+               onClose={Dialogs.close}
+               footer={
+                   <>
+                       <Button variant="primary"
+                               isLoading={isSaving}
+                               isDisabled={isSaving}
+                               onClick={saveSettings}>
+                           {_("Save changes")}
+                       </Button>
+                       <Button variant="link"
+                               isDisabled={isSaving}
+                               className="cancel"
+                               onClick={Dialogs.close}>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+               }>
+            {error && <ModalError dialogError={typeof error == 'string' ? error : error.message} />}
```

Comment:
This is too simplistic. I used separate CodeBlock for the details to keep the original line breaking of the journal. It's mnore ugly to read now:

![image](https://user-images.githubusercontent.com/200109/224905704-5fffdcc9-d2fd-4424-a4fe-37cf4ba38536.png)

It's fine to use some other element with pre-wrapped lines, but CodeBlock seemed easiest and most PF-ish.

Reply:
Ok, I can add that back. Not sure how it would fit in the dialogError, but I guess it can pass a `CodeBlock` as well.

Reply:
Added it back :)

Reply:
Is this something you'd like to see

![image](https://user-images.githubusercontent.com/67428/227575035-30a7eb7e-b317-4d35-97bb-5b189c0cce85.png)

Reply:
Yes, please! Ugh, did we actually land this PR with this formatting regression? Because I think it pretty much looked like that before, i.e. pre-wrapped lines.

Reply:
Yeah, it's not an ideal error message, but I think is the best we can do given the kdump error messaging being preformatted like this.

It might be good to strip off the redundant "kdump: " from the start of all of the lines though. But it's already a big improvement versus what's in main. Thanks!